benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java

```diff
@@ -196,12 +275,17 @@ private QueryExecutionResult executeSingleQuery(
         }
 
         macroService.runBenchmarkMacros(benchmark.getAfterExecutionMacros(), benchmark, connection);
-        return result;
+        return Optional.of(result);
     }
 
     private Connection getConnectionFor(QueryExecution queryExecution)
             throws SQLException
     {
         return applicationContext.getBean(queryExecution.getBenchmark().getDataSource(), DataSource.class).getConnection();
     }
+
+    private boolean isTimeLimitExceeded(Optional<ZonedDateTime> executionTimeLimit)
+    {
+        return executionTimeLimit.isPresent() && executionTimeLimit.get().compareTo(nowUtc()) < 0;
```

Comment:
use:
```
executionTimeLimit.ifPresent(limit -> ...).orElse(false);
```

Reply:
Actually, had to use map() as ifPresent doesn't return value.